src/mega-plusplus.coffee

```diff
@@ -55,10 +55,10 @@ module.exports = (robot) ->
     # the increment/decrement operator ++ or --
     (\+{2,}|-{2,})
     # optional reason for the plusplus
-    (?:\s+(?:#{reasonConjunctions})\s+(.+))?
-  ///i, (msg) ->
-    # let's get our local vars in place
-    [dummy, name, operator, reason] = msg.match
+    (?:\s+(?:#{reasonConjunctions})\s+([^@]+))?
```

Comment:
we don't currently  require @ to give karma. With this change to the regex, when using reasons, subsequent karmees only work when proceeded with @, otherwise the regex is greeedy and it gets picked up as part of the reason.

```
jetbot> zach+++ for bravery danny+++ for raisins
jetbot> zach has 7 points, 2 of which are for bravery danny+++ for raisins.
```

```
jetbot> zach+++ for bravery @danny+++ for raisins
jetbot> zach has 9 points, 2 of which are for bravery.
danny has 8 points, 2 of which are for raisins.
```

we can just accept that `@` is required for subsequent karmees when using reasons or we can try and get the regex to be less greedy (possibly madness inducing)

Reply:
This works now, but still has some limitations. Strings like you posted should give karma to the intended recipients now, but quoted strings won't work inside another statement.

Works:
```
"Boris the cat" ++++++ for being a good kitty other_kitty ++++++ for also being a good kitty
```

Doesn't work:
```
"Boris the cat" ++++++ for being a good kitty "Other kitty" ++++++ for also being a good kitty
```

If we can live with that limitation, this should do the trick.